### PR TITLE
Tune extensions for streams with variance annotation (out). (#13)

### PR DESCRIPTION
**Pull request by nostra13**

### CODE REVIEW

File: koptional-rxjava2-extensions/src/test/kotlin/com/gojuno/koptional/rxjava2/RxJava2ExtensionsSpec.kt [hunk 1]
```diff
@@ -124,7 +124,7 @@ class RxJava2ExtensionsSpec : Spek({
     context("Maybe") {
 
         context("Stream with Some") {
-            val stream by memoized {
+            val stream: Maybe<out Optional<String>> by memoized {
                 Maybe.just(Some("a") as Optional<String>)
```
Comment (artem-zinnatullin) [inline koptional-rxjava2-extensions/src/test/kotlin/com/gojuno/koptional/rxjava2/RxJava2ExtensionsSpec.kt]: So we should remove `as Optional<String>` then here and in similar places in tests

Comment (nostra13) [inline koptional-rxjava2-extensions/src/test/kotlin/com/gojuno/koptional/rxjava2/RxJava2ExtensionsSpec.kt]: Done

Comment (artem-zinnatullin) [inline koptional-rxjava2-extensions/src/test/kotlin/com/gojuno/koptional/rxjava2/RxJava2ExtensionsSpec.kt]: suk

Comment (nostra13) [inline koptional-rxjava2-extensions/src/test/kotlin/com/gojuno/koptional/rxjava2/RxJava2ExtensionsSpec.kt]: sidr

Comment (artem-zinnatullin) [inline koptional-rxjava2-extensions/src/test/kotlin/com/gojuno/koptional/rxjava2/RxJava2ExtensionsSpec.kt]: lox

**Comment by artem-zinnatullin** [comment]
@dmitry-novikov @ming13 what are we waiting for in this PR?

**Pull request merged**